device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/storage/timeSeries/TimeSeriesClient.java

```diff
@@ -97,37 +96,38 @@ public TimeSeriesClient(IServicesConfig config, WSClient wsClient)
     }
 
     @Override
-    public Status ping() {
-        String name = "TimeSeriesInsights";
+    public StatusResultServiceModel ping() {
+        StatusResultServiceModel result = new StatusResultServiceModel(false, "TimeSeriesInsights check failed");
+
         try {
             WSRequest request = this.PrepareRequest(
-                this.dataAccessFqdn,
-                AVAILABILITY_KEY,
-                this.acquireAccessToken());
+                    this.dataAccessFqdn,
+                    AVAILABILITY_KEY,
+                    this.acquireAccessToken());
 
-            WSResponse response = request
-                .get()
-                .toCompletableFuture()
-                .get();
+            WSResponse response = request.get().toCompletableFuture().get();
 
-            if (response.getStatus() == HttpStatus.SC_OK) {
-                return new Status(name, true, "Time Series Insights alive and well!");
+            if (response.getStatus() != 200) {
```

Comment:
should use pre-defined variable here (HttpStatus.SC_OK). Also, technically success is any 2xx status code

Reply:
IDK if we should we consider anything other than 200 as success.

Reply:
I did put predefined HTTPStatus, but should I check for 201, 202, 203 , 204 etc?